app/controllers/concerns/verify_sp_attributes_concern.rb

```diff
@@ -0,0 +1,49 @@
+module VerifySPAttributesConcern
+  def needs_verify_sp_attributes?
+    sp_session_identity.nil? || !requested_attributes_verified?
+  end
+
+  def verify_sp_attributes_url
```

Comment:
Since this method just delegates to `sign_up_completed_url`, we may should be able to remove it and just use `sign_up_completed_url`. I think that would make things easier on someone trying to figure out the code later.

Reply:
👍